src/overlays/actors/ovl_Eff_Dust/z_eff_dust.c

```diff
@@ -22,28 +36,349 @@ const ActorInit Eff_Dust_InitVars = {
+                        this->initialPositions[i].z = 2000.0f * Math_CosS(theta);
+                    }
+                    break;
+
+                case EFF_DUST_TYPE_4:
+                    this->initialPositions[i].x = (Rand_ZeroOne() * 8500.0f) + 1700.0f;
+                    if (this->initialPositions[i].x > 5000.0f) {
+                        this->initialPositions[i].y = (4000.0f * Rand_ZeroOne()) * Math_SinS(theta);
+                        this->initialPositions[i].z = (4000.0f * Rand_ZeroOne()) * Math_CosS(theta);
+                    } else {
+                        this->initialPositions[i].y = 4000.0f * Math_SinS(theta);
+                        this->initialPositions[i].z = 4000.0f * Math_CosS(theta);
+                    }
+
+                    break;
+                default:
+                    break;
+            }
+
+            this->distanceTraveled[i] = 0.0f;
+            this->index += 1;
+        }
+    }
+}
+
+void EffDust_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EffDust* this = THIS;
+
+    this->updateFunc(this, globalCtx);
+}
+
+void EffDust_DrawFunc_8099E4F4(Actor* thisx, GlobalContext* globalCtx2) {
+    EffDust* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Vec3f* initialPositions;
+    f32* distanceTraveled;
+    s32 i;
+    f32 aux;
+
+    OPEN_DISPS(gfxCtx, "../z_eff_dust.c", 425);
+
+    func_80093D18(gfxCtx);
+
+    gDPPipeSync(POLY_XLU_DISP++);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 128, 128, 128, 255);
+    gDPSetEnvColor(POLY_XLU_DISP++, 128, 128, 128, 0);
+
+    initialPositions = this->initialPositions;
+    distanceTraveled = this->distanceTraveled;
+
+    gSPSegment(POLY_XLU_DISP++, 0x08, D_8099EB60);
+
+    for (i = 0; i < 0x40; i++) {
+        if (*distanceTraveled < 1.0f) {
+            // Needed to match.
+            if (!this) {}
+
+            aux = 1.0f - (*distanceTraveled * *distanceTraveled);
+            Matrix_Translate(this->actor.world.pos.x + (initialPositions->x * ((this->dx * aux) + (1.0f - this->dx))),
+                             this->actor.world.pos.y + (initialPositions->y * ((this->dy * aux) + (1.0f - this->dy))),
+                             this->actor.world.pos.z + (initialPositions->z * ((this->dz * aux) + (1.0f - this->dz))),
+                             MTXMODE_NEW);
+
+            Matrix_Scale(this->scalingFactor, this->scalingFactor, this->scalingFactor, MTXMODE_APPLY);
+            Matrix_Mult(&globalCtx->mf_11DA0, MTXMODE_APPLY);
+
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx, "../z_eff_dust.c", 449),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(gEffSparklesDL));
+        }
+
+        initialPositions++;
+        distanceTraveled++;
+        // Needed for matching.
+        if (1) {}
+        {}
+        if (1) {}
+        if (1) {}
+        if (1 != 0) {}
```

Comment:
Are all of these needed to match? And if they are, can they simply be `if (1)`s by any chance?